app/models/infra_conversion_job.rb

```diff
@@ -33,18 +29,31 @@ def load_transitions
     self.state ||= 'initialize'
 
     {
-      :initializing     => {'initialize'       => 'waiting_to_start'},
-      :start            => {'waiting_to_start' => 'running'},
-      :poll_conversion  => {'running'          => 'running'},
-      :start_post_stage => {'running'          => 'post_conversion'},
-      :poll_post_stage  => {'post_conversion'  => 'post_conversion'},
-      :finish           => {'*'                => 'finished'},
-      :abort_job        => {'*'                => 'aborting'},
-      :cancel           => {'*'                => 'canceling'},
-      :error            => {'*'                => '*'}
+      :initializing                => {'initialize'       => 'waiting_to_start'},
+      :start                       => {'waiting_to_start' => 'started'},
+      :poll_automate_state_machine => {
+        'started'             => 'running_in_automate',
+        'running_in_automate' => 'running_in_automate'
+      },
+      :finish                      => {'*'                => 'finished'},
+      :abort_job                   => {'*'                => 'aborting'},
+      :cancel                      => {'*'                => 'canceling'},
+      :error                       => {'*'                => '*'}
     }
   end
 
+  def load_states
+    {
+      :running_in_automate => {
+        :max_retries => 8640 # 36 hours with a retry interval of 15 seconds
```

Comment:
I don't think this belongs in the states hash as it doesn't describe the state transition, wouldn't this belong in Job.options and/or Settings

Reply:
The `states` hash doesn't exist in the `Job` class, so I created it to describe the state itself, not the transitions. And I mimic'd the `load_transitions` method for that. The idea is to store information that will be used to control the state (`max_retries`), and to update the task (`description`).

We could put everything in the Settings, but that would mean a lot of information there. And I liked the idea of the self-contained transitions hash.

Reply:
Oh `@state` vs `@states` is a little confusing, I thought you were modifying https://github.com/ManageIQ/manageiq/blob/10307727b4394b6737a66e220abff1953e1c3f7e/app/models/infra_conversion_job.rb#L29 but that ends up going to `@transitions`

I like the idea of per-state settings, but I think it should be able to be modified at runtime which doesn't really fit with the model of statically defined transitions, this is the purpose of the options hash.

Reply:
Well, in my mind, the `states` hash was meant to be static, like `transitions`.
For dynamic content, I'm using `job.context`, like https://github.com/ManageIQ/manageiq/pull/19164/files#diff-51b91017cbf3d1648a52ca0482a63650R78.

Reply:
I think the likelihood of someone wanting to change max_retries is quite high :)

Reply:
Well, there's already a max_retries in the Automate implementation and we've got no feedback from people who want to change it. We've put a _high enough_ value, that should prevent timeout, apart from really bad situations.

We could still put that in Settings, but shouldn't we also put the transitions ?

Reply:
Well transitions are static, how long something takes to run is environmental.

Reply:
I'm okay with having this static, even though there's almost always a customer for which a statically defined timeout is wrong.  Biggest issue is I think the name of this `states` is not descriptive enough.  If you named this `state_settings` that would be good enough, and I don't think you need the `load_states` part this could just a static instance var.

Reply:
I updated the code. Does it suits you better ?
And if we change our mind, we can easily revisit it later :smiley:

Reply:
Well states was the name I had an issue with, and wanted to get away from load_states so this is the opposite of what I was thinking haha

Reply:
My bad. I didn't refresh the Github page, so the last comment was https://github.com/ManageIQ/manageiq/pull/19164#discussion_r315361700

Reply:
:+1:  no worries